crates/subspace-networking/src/peer_info/handler.rs

```diff
@@ -48,37 +49,30 @@ impl Config {
 /// The successful result of processing an inbound or outbound peer info requests.
 #[derive(Debug)]
 pub enum PeerInfoSuccess {
+    /// Local peer received peer info from a remote peer.
     Received(PeerInfo),
-    DataSent,
+    /// Local peer sent its peer info to a remote peer.
+    Sent,
 }
 
 /// A peer info protocol failure.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum PeerInfoError {
     /// The peer does not support the peer info protocol.
+    #[error("Peer info protocol is not supported.")]
     Unsupported,
     /// The peer info request failed.
+    #[error("Peer info error: {error}")]
     Other {
+        #[source]
         error: Box<dyn Error + Send + 'static>,
     },
 }
 
-impl fmt::Display for PeerInfoError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            PeerInfoError::Other { error } => write!(f, "Peer info error: {error}"),
-            PeerInfoError::Unsupported => write!(f, "Peer info protocol not supported"),
-        }
-    }
-}
-
-impl Error for PeerInfoError {
-    fn source(&self) -> Option<&(dyn Error + 'static)> {
-        match self {
-            PeerInfoError::Other { error } => Some(&**error),
-            PeerInfoError::Unsupported => None,
-        }
-    }
+/// Marker struct for outbound peer-info requests.
```

Comment:
Nit about naming: markers don't contain data, they are not a payload themselves either. Markers are usually either traits or structs with no fields used in generics.